flang/lib/Lower/Bridge.cpp

```diff
@@ -1473,7 +1473,25 @@ class FirConverter : public Fortran::lower::AbstractConverter {
   /// the global to the ModuleOp as a new uniqued symbol and initialize it with
   /// the correct value. It will be referenced on demand using `fir.addr_of`.
   void instantiateGlobal(const Fortran::lower::pft::Variable &var) {
-    llvm_unreachable("Varun: put your code here");
+    const auto &sym = var.getSymbol();
+    std::string globalName = uniqueCGIdent(sym.name().ToString());
+    auto global = builder->getNamedGlobal(globalName);
+    if (!global) {
+      if (const auto *details =
+              sym.detailsIf<Fortran::semantics::ObjectEntityDetails>()) {
+        if (details->init()) {
+          global = builder->createGlobal(
+              toLocation(), genType(sym), globalName, false,
+              [&](Fortran::lower::FirOpBuilder &builder) {
+                auto initVal = genExprValue(details->init().value());
+                builder.create<fir::HasValueOp>(toLocation(), initVal);
+              });
+          auto addrOf = builder->create<fir::AddrOfOp>(
+              toLocation(), global.resultType(), global.getSymbol());
+          addSymbol(sym, addrOf);
+        }
+      }
```

Comment:
What happens if the global does not have an initializer?

Reply:
This case isn't handled. Peter said that all compilers employ an extension to the standard and initialize globals to zero. I will try to do the same here.

Reply:
You just have to instantiate an external global. The initializer (to zero) is the default and happens for free.